sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java

```diff
@@ -933,4 +945,210 @@ public void shouldNotOperateOnLocalStorageWhenPartitionNotExist() {
         assertNotNull(deleteDocument.getDocumentError());
         assertTrue(deleteDocument.getDocumentError().getError().getMessage().contains(failedMessage));
     }
+
+    @Test
+    public void pendingOperationProcessedWhenNetworkOnAndApplyAppEnabled() throws JSONException {
+        final PendingOperation pendingOperation = new PendingOperation(
+                USER_TABLE_NAME,
+                PENDING_OPERATION_DELETE_VALUE,
+                RESOLVED_USER_PARTITION,
+                DOCUMENT_ID,
+                "document",
+                BaseOptions.DEFAULT_EXPIRATION_IN_SECONDS);
+        when(mNetworkStateHelper.isNetworkConnected()).thenReturn(true);
+        when(mLocalDocumentStorage.getPendingOperations(USER_TABLE_NAME)).thenReturn(
+                new ArrayList<PendingOperation>() {{
+                    add(pendingOperation);
+                }});
+        ArgumentCaptor<DocumentMetadata> documentMetadataArgumentCaptor = ArgumentCaptor.forClass(DocumentMetadata.class);
+
+        Storage.setDataStoreRemoteOperationListener(mDataStoreEventListener);
+
+        mStorage.applyEnabledState(true);
```

Comment:
Avoid bypassing public APIs, any reason Storage.setEnabled(true).get() does not work?

Reply:
Set it to false and true works!